### PR TITLE
CXP-1333: Be able to count the number of products in the selection criteria

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/front/src/components/ProductSelection/ProductSelection.tsx [hunk 1]
```diff
@@ -24,6 +25,7 @@ type Props = {
 
 const ProductSelection: FC<Props> = ({criteria, onChange, errors}) => {
     const [values, dispatch] = useReducer(ProductSelectionReducer, criteria);
+    const debouncedValue = useDebounce<ProductSelectionValues>(values, 2000);
```
Comment (hogren) [inline components/catalogs/front/src/components/ProductSelection/ProductSelection.tsx:28]: The variable name is not clear. At this step, we do not know the goal of it. We just know that the variable use a debounce method.

File: components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const {data: count} = useCountProductsInSelectionCriteria(criteria);
+
+    let label = '';
+    if (null === count || undefined === count) {
+        label = translate('akeneo_catalogs.product_selection.count.error');
+    } else if (0 === count) {
+        label = count.toString() + ' ' + translate('akeneo_catalogs.product_selection.count.no_products');
+    } else {
+        label = count.toString() + ' ' + translate('akeneo_catalogs.product_selection.count.products');
+    }
```
Comment (hogren) [inline components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx:35]: Can you use a more concise translate method, like at line 117 of `Akeneo/Connectivity/Connection/front/src/connect/components/Marketplace.tsx` (associated with line 60 of `src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml`) ?

File: components/catalogs/front/src/components/ProductSelection/ProductSelection.tsx [hunk 3]
```diff
@@ -38,6 +40,7 @@ const ProductSelection: FC<Props> = ({criteria, onChange, errors}) => {
     return (
         <ProductSelectionContext.Provider value={dispatch}>
             <Header>
+                <CountSelectedProductsByCriteria criteria={debouncedValue} />
```
Comment (guillaumefer-akeneo) [inline components/catalogs/front/src/components/ProductSelection/ProductSelection.tsx:43]:
```suggestion
                <CountSelectedProductsByCriteria criteria={debouncedCriteria} />
```
`criteria={debouncedValue}` is hard to understand IMO

File: components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+export const CountSelectedProductsByCriteria: FC<Props> = ({criteria}) => {
+    const translate = useTranslate();
+    const {data: count} = useCountProductsInSelectionCriteria(criteria);
+
+    let label = '';
+    if (null === count || undefined === count) {
+        label = translate('akeneo_catalogs.product_selection.count.error');
+    } else if (0 === count) {
+        label = count.toString() + ' ' + translate('akeneo_catalogs.product_selection.count.no_products');
```
Comment (guillaumefer-akeneo) [inline components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx:32]: `0 No products` ?
```suggestion
        label = translate('akeneo_catalogs.product_selection.count.no_products');
```

Comment (guillaumefer-akeneo) [inline components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx:32]: `0 No products` ?
```suggestion
        label = translate('akeneo_catalogs.product_selection.count.no_products');
```

**Review by guillaumefer-akeneo — APPROVED** [review]

**Review by eoliveauakeneo — APPROVED** [review]

File: components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+const Count = styled.span`
+    color: ${getColor('brand', 100)};
+    font-size: ${getFontSize('default')};
+`;
+
+type Props = {
+    criteria: ProductSelectionValues;
+};
+
+export const CountSelectedProductsByCriteria: FC<Props> = ({criteria}) => {
```
Comment (masacc) [inline components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx:24]: can it be a good idea to memoize this component?

File: components/catalogs/back/src/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 6]
```diff
@@ -38,6 +38,10 @@ akeneo_catalogs:
                 placeholder: Select a currency
     product_selection:
         empty: Add selection criteria to create your product selection
+        count:
+            products: products
+            no_products: No products
+            error: Impossible to calculate the number of products
```
[the remark this answers is not in the document]
Comment (willy-ahva) [reply components/catalogs/back/src/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml:44]: I copy pasted from figma. There is no dot.

File: components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx (hunk 2 above)
Comment (willy-ahva) [reply components/catalogs/front/src/components/ProductSelection/components/CountSelectedProductsByCriteria.tsx:35]: totally agree!

**Pull request merged**